openapi-core/src/main/java/io/ballerina/openapi/core/generators/common/GeneratorUtils.java

```diff
@@ -767,8 +768,9 @@ public static OpenAPI normalizeOpenAPI(OpenAPI openAPI, boolean validateOpIds, b
         }
         validateRequestBody(openAPIPaths.entrySet());
         if (isSanitized) {
-            OASModifier oasSanitizer = new OASModifier(openAPI);
-            return oasSanitizer.modifyWithBallerinaConventions();
+            OASModifier oasSanitizer = new OASModifier();
+            Optional<Map<String, String>> proposedNameList = oasSanitizer.getProposedNameList(openAPI);
+            return oasSanitizer.modifyWithBallerinaConventions(openAPI, proposedNameList.get());
```

Comment:
Don't we need to check whether the optional `proposedNameList` is empty before calling the `get`

Reply:
Yeah it was added with this https://github.com/ballerina-platform/openapi-tools/pull/1762/commits/deca2c23959600fefc619afbde162eecec1036f0